numpy/__init__.pyi

```diff
@@ -386,133 +388,172 @@ class generic(_ArrayOrScalarCommon):
     @property
     def base(self) -> None: ...
 
-class _real_generic(generic):  # type: ignore
+class number(generic):  # type: ignore
     @property
     def real(self: _ArraySelf) -> _ArraySelf: ...
     @property
     def imag(self: _ArraySelf) -> _ArraySelf: ...
 
-class number(generic): ...  # type: ignore
-
-class bool_(_real_generic):
+class bool_(generic):
     def __init__(self, __value: object = ...) -> None: ...
+    @property
+    def real(self: _ArraySelf) -> _ArraySelf: ...
+    @property
+    def imag(self: _ArraySelf) -> _ArraySelf: ...
 
 class object_(generic):
     def __init__(self, __value: object = ...) -> None: ...
+    @property
+    def real(self: _ArraySelf) -> _ArraySelf: ...
+    @property
+    def imag(self: _ArraySelf) -> _ArraySelf: ...
 
-class datetime64:
+class datetime64(generic):
     @overload
     def __init__(
         self,
-        __value: Union[None, datetime64, str, dt.datetime] = ...,
-        __format: str = ...
+        __value: Union[None, datetime64, str, bytes, dt.datetime] = ...,
+        __format: Union[str, bytes] = ...,
```

Comment:
```suggestion
        __format: Union[str, bytes, Tuple[Union[str, bytes], int]] = ...,
```

Reply:
Fixed in https://github.com/numpy/numpy/commit/6a3bdeb81e3cc7530d10269f09c414c874f44866.